api/src/main/java/org/pmiops/workbench/cohortbuilder/SearchGroupItemQueryBuilder.java

```diff
@@ -0,0 +1,583 @@
+      if (param.getAttributes().isEmpty()) {
+        childConceptIds.addAll(childConceptIds(criteriaLookup, ImmutableList.of(param)));
+        // make sure we only add the concept ids sql template once
+        if (!queryParts.contains(CONCEPT_ID_IN)) {
+          queryParts.add(CONCEPT_ID_IN);
+        }
+      } else {
+        StringBuilder bpSql = new StringBuilder(BP_SQL);
+        List<Long> bpConceptIds = new ArrayList<>();
+        for (Attribute attribute : param.getAttributes()) {
+          if (attribute.getConceptId() != null) {
+            // this indicates we need to build a blood pressure sql statement
+            bpConceptIds.add(attribute.getConceptId());
+            processBloodPressureSql(queryParams, bpSql, attribute);
+          } else if (AttrName.NUM.equals(attribute.getName())) {
+            queryParts.add(processNumericalSql(queryParams, param, attribute));
+          } else {
+            queryParts.add(processCategoricalSql(queryParams, param, attribute));
+          }
+        }
+        if (!bpConceptIds.isEmpty()) {
+          // if blood pressure we need to add named parameters for concept ids
+          QueryParameterValue cids =
+              QueryParameterValue.array(bpConceptIds.stream().toArray(Long[]::new), Long.class);
+          String conceptIdsParam = addQueryParameterValue(queryParams, cids);
+          queryParts.add(String.format(bpSql.toString(), conceptIdsParam) + ")\n");
+        }
+      }
+      // all search parameters in the search group item have the same source/standard flag
+      isStandard = param.getStandard();
+    }
+    // create the source/standard flag named parameter
+    String standardParam =
+        addQueryParameterValue(queryParams, QueryParameterValue.int64(isStandard ? 1 : 0));
+    // need to add standard flat to the sql statement
+    String queryPartsSql = String.format(STANDARD_SQL, standardParam);
+    // need to OR all query parts together since they exist in the same search group item
+    queryPartsSql = queryPartsSql + "(" + String.join(OR, queryParts) + ")\n";
+    if (!childConceptIds.isEmpty()) {
+      // if we have concept ids that are non bp then add named parameters
+      QueryParameterValue cids =
+          QueryParameterValue.array(childConceptIds.stream().toArray(Long[]::new), Long.class);
+      String conceptIdsParam = addQueryParameterValue(queryParams, cids);
+      queryPartsSql = String.format(queryPartsSql, conceptIdsParam);
+    }
+    // format the base sql with all query parts
+    String baseSql = BASE_SQL + queryPartsSql;
+    // build modifier sql if modifiers exists
+    String modifiedSql = buildModifierSql(baseSql, queryParams, searchGroupItem.getModifiers());
+    // build the inner temporal sql if this search group item is temporal
+    // otherwise return modifiedSql
+    return buildInnerTemporalQuery(
+        modifiedSql, queryPartsSql + "%s", queryParams, searchGroupItem.getModifiers(), mention);
```

Comment:
Could you add a comment to explain this `%s`?

Reply:
I removed it, not needed.